p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -390,50 +390,64 @@ public boolean addProtectedStorageEntry(ProtectedStorageEntry protectedStorageEn
             return false;
         }
 
-        boolean sequenceNrValid = isSequenceNrValid(protectedStorageEntry.getSequenceNumber(), hashOfPayload);
-        boolean result = sequenceNrValid &&
-                checkPublicKeys(protectedStorageEntry, true)
-                && checkSignature(protectedStorageEntry);
+        // TODO: Combine with hasSequenceNrIncreased check, but keep existing behavior for now
+        if(!isSequenceNrValid(protectedStorageEntry.getSequenceNumber(), hashOfPayload) ) {
+            log.trace("addProtectedStorageEntry failed due to invalid sequence number");
+
+            return false;
+        }
+
+        // Verify the ProtectedStorageEntry is well formed and valid for the add operation
+        if (!checkPublicKeys(protectedStorageEntry, true) ||
+                !checkSignature(protectedStorageEntry)) {
+
+            log.trace("addProtectedStorageEntry failed due to invalid entry");
+            return false;
```

Comment:
Maybe better to split that to 2 checks and more concrete log messages.

Reply:
The helper methods already have verbose logging and each case will print out a separate message. I updated this PR to only add new messages for the failures that didn't have them before. It is worth another pass to clean these up again, but I don't want to pollute this stack with those changes.